Wikipedia/UI-V5/WMFNearbyViewModel.m

```diff
@@ -0,0 +1,193 @@
+    [self.autoUpdatingIndexes removeIndex:index];
+    DDLogInfo(@"Tracking headings for %lu indexes", self.autoUpdatingIndexes.count);
+    // TODO: toggle heading tracking based on whether or not we have indexes to track
+}
+
+- (void)stopUpdatingAllResults {
+    [self.autoUpdatingIndexes enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL* stop) {
+        [self stopUpdatingResultAtIndex:idx];
+    }];
+}
+
+- (void)updateBearingAtIndex:(NSUInteger)index {
+    MWKLocationSearchResult* result = self.locationSearchResults.results[index];
+    result.bearingToLocation =
+        [self.locationSearchResults.location wmf_bearingToLocation:result.location
+                                                 forCurrentHeading:self.locationManager.lastHeading];
+}
+
+- (void)updateDistanceAtIndex:(NSUInteger)index {
+    MWKLocationSearchResult* result = self.locationSearchResults.results[index];
+    result.distanceFromUser = [result.location distanceFromLocation:self.locationManager.lastLocation];
+}
+
+#pragma mark - Fetch
+
+- (void)startUpdates {
+    [self.locationManager restartLocationMonitoring];
+    // TODO: check if authorized. if not, fail w/ an error
+}
+
+- (void)stopUpdates {
+    [self.locationManager stopMonitoringLocation];
+    self.locationSearchResults = nil;
+    // TODO: cancel last search request
+}
+
+- (void)fetchTitlesForLocation:(CLLocation* __nullable)location {
+    if (!location) {
+        return;
+    } else if ([self.locationSearchResults.location distanceFromLocation:location] < 500
+               && [self.locationSearchResults.searchSite isEqualToSite:self.site]) {
+        /*
+           How often we search titles for a location is separate from how often the location updates.
+           See WMFLocationManager for distanceFilter settings.
+         */
+        DDLogVerbose(@"Not fetching nearby titles for %@ since it is too close to previously fetched location: %@.",
+                     location, self.locationSearchResults.location);
+        return;
+    }
+
+    NSAssert(![self.locationSearchResults.location wmf_isVeryCloseToLocation:location]
+             || self.locationSearchResults.searchSite,
+             @"Unexpected fetch of two nearly adjacent locations: %@", @[self.locationSearchResults.location, location]);
+
+    // TODO: cancel fetch for previous location
+    @weakify(self);
+    [self.locationSearchFetcher fetchArticlesWithSite:self.site location:location resultLimit:self.resultLimit]
+    .then(^(WMFLocationSearchResults* locationSearchResults) {
+        @strongify(self);
+        if (![locationSearchResults.searchSite isEqualToSite:self.site]) {
+            DDLogWarn(@"Ignoring location search results from site %@ since it was changed to %@",
+                      locationSearchResults.searchSite, self.site);
+            return;
+        } else if (![locationSearchResults.location wmf_hasSameCoordinatesAsLocation:self.locationManager.lastLocation]) {
```

Comment:
I see what you're saying now. That second conditional isn't supposed to be there.  I was thinking it would be like how we manage cancellation logic for `MWKImage`, but it's not applicable here since we don't have an accurate representation of the "last requested location"—though we could.